app/models/customer.rb

```diff
@@ -9,4 +9,11 @@ def self.top_five_completed_transactions
     .group(:id)
     .limit(5)
   end
+
+  def self.top_5_customers
+    joins(invoices: :transactions)
+    .select("customers.*, COUNT(distinct transactions.id) as transaction_count")
+    .where("transactions.result = 1")
+    .group("customers.id").order("transaction_count desc").limit(5)
```

Comment:
I just learned today that once you create a custom column, you can call it as a symbol, like `order(transaction_count: :desc)`

Reply:
And in your `group` statement, I *believe* you should just be able to `group(:id)` right? Since I think it will default to the table you're starting with. Could be wrong here, but worth a try?

Reply:
Just tried both, looks like they both work!!